sshd-common/src/main/java/org/apache/sshd/common/config/keys/OpenSshCertificate.java

```diff
@@ -0,0 +1,262 @@
+        private PublicKey serverHostKey;
+        private long serial;
+        private int type;
+        private String id;
+        private List<String> principals;
+        private long validAfter;
+        private long validBefore;
+        private List<String> criticalOptions;
+        private List<String> extensions;
+        private String reserved;
+        private PublicKey caPubKey;
+        private byte[] message;
+        private byte[] signature;
+
+        private OpenSshPublicKeyBuilder() {
+        }
+
+        public static OpenSshPublicKeyBuilder anOpenSshCertificate() {
+            return new OpenSshPublicKeyBuilder();
+        }
+
+        public OpenSshPublicKeyBuilder withKeyType(String keyType) {
+            this.keyType = keyType;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withNonce(byte[] nonce) {
+            this.nonce = nonce;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withServerHostPublicKey(PublicKey serverHostKey) {
+            this.serverHostKey = serverHostKey;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withSerial(long serial) {
+            this.serial = serial;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withType(int type) {
+            this.type = type;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withId(String id) {
+            this.id = id;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withPrincipals(List<String> principals) {
+            this.principals = principals;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withValidAfter(long validAfter) {
+            this.validAfter = validAfter;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withValidBefore(long validBefore) {
+            this.validBefore = validBefore;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withCriticalOptions(List<String> criticalOptions) {
+            this.criticalOptions = criticalOptions;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withExtensions(List<String> extensions) {
+            this.extensions = extensions;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withReserved(String reserved) {
+            this.reserved = reserved;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withCaPubKey(PublicKey caPubKey) {
+            this.caPubKey = caPubKey;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withMessage(byte[] message) {
+            this.message = message;
+            return this;
+        }
+
+        public OpenSshPublicKeyBuilder withSignature(byte[] signature) {
+            this.signature = signature;
+            return this;
+        }
+
+        public OpenSshCertificate build() {
```

Comment:
Since the only way to build an instantiated class is via the builder and there are only getters the fields should be `final`

Reply:
How can I do this without having all parameters in the constructor arguments? Again the reason for the builder is:
`Error:(47,13) java: (sizes) ParameterNumber: More than 8 parameters (found 15).`

Reply:
You can pass the builder instance to the constructor